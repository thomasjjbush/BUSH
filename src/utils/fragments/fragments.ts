import { gql } from 'graphql-request';

export const clientFragment = gql`
    fragment ClientFragment on Clients {
        name
        primaryColor
        logo {
            url
        }
        slug
        url
    }
`;

export const projectFragment = gql`
    fragment ProjectFragment on Project {
        client {
            name
            primaryColor
            logo {
                url
            }
            slug
            url
        }
        name
        slug
        tags: tagsCollection(limit: 20) {
            items {
                name
                slug
            }
        }
    }
`;

export const similarFragment = gql`
    fragment SimilarFragment on Project {
        name
        slug
        hero {
            url
        }
    }
`;

export const tagFragment = gql`
    fragment TagFragemnt on Tag {
        name
        slug
    }
`;
