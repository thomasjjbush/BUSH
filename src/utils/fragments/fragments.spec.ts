import { clientFragment, projectFragment, similarFragment, tagFragment } from './fragments';

describe('fragments', () => {
    describe('clientFragment', () => {
        it('should match snapshot', () => {
            expect(clientFragment).toMatchSnapshot();
        });
    });

    describe('projectFragment', () => {
        it('should match snapshot', () => {
            expect(projectFragment).toMatchSnapshot();
        });
    });

    describe('similarFragment', () => {
        it('should match snapshot', () => {
            expect(similarFragment).toMatchSnapshot();
        });
    });

    describe('tagFragmemt', () => {
        it('should match snapshot', () => {
            expect(tagFragment).toMatchSnapshot();
        });
    });
});
