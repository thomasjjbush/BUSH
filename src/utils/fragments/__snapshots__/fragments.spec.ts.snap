// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fragments clientFragment should match snapshot 1`] = `
"
    fragment ClientFragment on Clients {
        name
        primaryColor
        logo {
            url
        }
        slug
        url
    }
"
`;

exports[`fragments projectFragment should match snapshot 1`] = `
"
    fragment ProjectFragment on Project {
        client {
            name
            primaryColor
            logo {
                url
            }
            requiresInverseLogo
            slug
            url
        }
        name
        slug
        tags: tagsCollection(limit: 20) {
            items {
                name
                slug
            }
        }
    }
"
`;

exports[`fragments similarFragment should match snapshot 1`] = `
"
    fragment SimilarFragment on Project {
        name
        slug
        hero {
            url
        }
    }
"
`;

exports[`fragments tagFragmemt should match snapshot 1`] = `
"
    fragment TagFragemnt on Tag {
        name
        slug
    }
"
`;
